rust_src/src/windows.rs

```diff
@@ -1031,4 +1031,12 @@ pub fn window_new_total(arg: LispObject) -> LispObject {
     win.new_total
 }
 
+/// Return the number of columns by which WINDOW is scrolled from left margin.
+/// WINDOW must be a live window and defaults to the selected one.
+#[lisp_fn(min = "0")]
+pub fn window_hscroll(window: LispObject) -> LispObject {
+    let window = window_live_or_selected(window);
```

Comment:
Oh. Gotcha :+1:

Reply:
yeah, it is annoying because the compiler says `i64`. It does not know to show the type alias.

Reply:
theoretically it does not matter, but we used to worry about 32-bit support. `EmacsInt` helped us abstract that.

Reply:
Right, and I was also puzzled if this is a sane choice in the first place, but good to know, thanks!

Reply:
And I was clearly tired. This can be simplified further. Sorry about that.

```
#[lisp_fn(min = "0")]
pub fn window_hscroll(window: LispObject) -> EmacsInt {
    let window = window_live_or_selected(window);
    window.hscroll as EmacsInt
}

Reply:
Ah, that's neat! I pushed the change

Reply:
yeah, as the docs discuss, the `lisp_fn` macro uses the `From` trait to convert LispObjects into Rust types. and it does the same on the Rust type used for return. *ALL* lisp_fn functions end up taking LispObject and returning LispObject. But it is nicer for us to work with Rust objects.

In the C code the DEFUN becomes `Ffunction_name` from `DEFUN ("function-name"....)`. So you will find calls like `Fcons(....)` or `Fdo_thing(....)` in the C. Any of the Rust code that calls `Fdo_thing` is calling the C code.

`lisp_fn` takes the Rust function and creates a wrapper function for it called `Ffunction_name` but leaves the original `function_name` behind as a standard Rust function. So in Rust we can call `function_name(....)` or `Ffunction_name(...)` depending on whether we have LispObjects (Ffunction_name) or Rust object (function_name).

Reply:
The `min = "0"` means that the LispObject can be Qnil and is translated as `Option<SomeTime>` in Rust.

Reply:
That makes sense. Thank you for the explanation!